### ISSUE TITLE
train on different object classes 

### ISSUE THREAD

**Issue by aminemosbah** [matterport/Mask_RCNN#1901]
tried to train on my own classes, following balloon tutorial but I still get balloon label on my object , any suggestion to get it right

**Comment by marcfielding1** [comment]
Yeah edit your class to label it correctly so I reckon roughly...

https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/samples/balloon/balloon.py#L90

and here

https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/samples/balloon/balloon.py#L137

You need to actually read the balloon tutorial buddy and understand what the classes do!

**Comment by aminemosbah** [comment]
config.CLASS_NAMES))
      6     for id, region in a['regions']:
----> 7         object_name = region['region_attributes']['name']
      8         uniq_class_names[object_name] += 1
      9         total_classes += 1

TypeError: string indices must be integers

raises
when adding .a['regions'].items()
it raises an error: no item 
any proposal?
thanks

**Comment by aminemosbah** [comment]
actually, I get stuck in configuration bug when trying to test on my own data
it s related to the configuration file for loading and preparing data when I tried to overwrite the balloon configuration to my actual data it breaks, 
I m training on collab
thanks

**Comment by aminemosbah** [comment]
hi, great tutorial however ai get stuck in configuration bug when trying to test on my own data
here is the bug

TypeError                                 Traceback (most recent call last)
<ipython-input-46-4e82ffc8d52d> in <module>()
      5     rc = np.zeros(len(config.CLASS_NAMES))
      6     for id, region in a['regions']:
----> 7         object_name = region['region_attributes']['name']
      8         uniq_class_names[object_name] += 1
      9         total_classes += 1

TypeError: string indices must be integers

when adding .a['regions'].items()
it raises an error: no item 
any proposal?
thanks

**Comment by aashokvardhan** [comment]
Looks like 'region' is a string over here.
for i, region in a['regions']:
       print(i)
       print(region)

output:
region_attributes
shape_attributes

instead use:
for region in a['regions']:
      object_name = region['region_attributes']['name']

**Comment by nataliameira** [comment]
@aminemosbah  I have the same problem. Has anyone managed to solve this?

**Comment by marcfielding1** [comment]
> Yeah edit your class to label it correctly so I reckon roughly...
> 
> https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/samples/balloon/balloon.py#L90
> 
> and here
> 
> https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/samples/balloon/balloon.py#L137
> 
> You need to actually read the balloon tutorial buddy and understand what the classes do!

@nataliameira That should fix it.

**Comment by nataliameira** [comment]
Hi @marcfielding1

Thank you for your answer. My error is the same as the one mentioned by @aminemosbah:

`TypeError: string indices must be integers`

Can you help me? I'm working with 5 classes in the balloon example. I saw other people with the same problem, but I couldn't find a solution.

Your help will be appreciated.

**Comment by marcfielding1** [comment]
Heya sorry, I didn't catch that reply with the type error, my bad!

Hmm - give me half an hour let me grab my personal repo and see what I did! What file do you get that error in please(post the stack trace)?

**Comment by nataliameira** [comment]
hey @marcfielding1 

I solved my problem. The problem was reading my .json file with the annotations.

Thank you very much for your help!
